tests/subsys/llext/testcase.yaml

```diff
@@ -142,3 +142,20 @@ tests:
       - CONFIG_LLEXT_STORAGE_WRITABLE=n
       - CONFIG_LLEXT_EXPORT_DEV_IDS_BY_HASH=y
       - CONFIG_LLEXT_EXPORT_BUILTINS_BY_SLID=y
+
+  # Test at least one platform and arch for dynamically linked extensions
+  # which set up lookup tables to modify.
+  # NOTE: note all toolchains/architectures support this, and if you want to test it
+  # add your *specific* platforms here
+  llext.elf_sharedlib:
+    arch_allow:
+      - xtensa
+    toolchain_exclude:
+      - xt-clang
+    integration_platforms:
+      - qemu_xtensa/dc233c      # Xtensa ISA
+    filter: not CONFIG_MPU and not CONFIG_MMU
+    extra_conf_files: ['no_mem_protection.conf']
+    extra_configs:
+      - CONFIG_LLEXT_STORAGE_WRITABLE=y
+      - CONFIG_LLEXT_TYPE_ELF_SHAREDLIB=y
```

Comment:
This new entry is now replacing the previous meaning of `llext.writable` on Xtensa, while `llext.writable_relocatable` now becomes identical to `llext.writable` there, same for SLID versions. Maybe we shouldn't rely on arch defaults and make explicit tests for each ELF type? E.g. `llext.writable_relocatable`, `llext.writable_sharedlib` and `llext.writable_object`?

Reply:
I'm frankly not even sure what the object elf type is here... there's no ET_OBJECT or ET_SHARED... its ET_REL or ET_DYN that we actually support here and realistically I need to correct this.

Reply:
I plan on fixing the object type options in the future. I don't want to break things right before a release.

Reply:
Sorry, I still don't think having `llext.writable` and `llext.writable_relocatable` perform the same testing on Xtensa (and potentially on other platforms) is a particularly good option. What should `llext.writable` now mean? Don't think leaving it just the default is very beneficial here since it can overlap with other tests